### ISSUE TITLE
Estandarizar la creación de Pull Requests

### ISSUE THREAD

**Issue by limcross** [Ticketplus/guides#9]
## Descripción
Es necesario establecer una única forma de presentar los cambios, facilitando de esta forma la revisión de los cambios, y que no solo se traduzca en una revisión de estilo de código.

## Posible Solución
Crear una plantilla que pida los elementos necesarios para comprender el alcance de la solución y las acciones, que sean necesarias para revisar los cambios y/o llevarlos a producción.

* **Formato de título**: [jira-link] Título
* **Contenido del Pull Request**:
  * **Vista previa**:
  En caso de que el cambio incluye la adición de vistas o edición de estas, deberá ser incluida una imagen que describa los flujos de la solución.
  * **Descripción**:
  Descripción breve de la solución entregada.
  * **Requerimientos**:
  De ser necesarios aquí se indicarán los requisitos previos para probar y/o desplegar los cambios.

**Comment by maxilopezramirez** [comment]
Creo que deberíamos aprovechar la nomenclatura que ya dispone jira al momento de crear remas desde un ticket.

**Comment by rmartineztp** [comment]
Estoy de acuerdo con la solución que propone @limcross, me parece buenísimo lo de la vista previa, en algún momento lo hice en los comentarios de algunos PR pero me dijeron que ese tipo de cosas las colocara en la respectiva tarjeta de jira, algo que me pareció muy bien porque en ocaciones los cambios pueden ser tantos que dejarlos en la vista previa del PR lleva como mucha información y lo más importante del PR son los cambios en el código.

**Comment by limcross** [comment]
> Creo que deberíamos aprovechar la nomenclatura que ya dispone jira al momento de crear ramas desde un ticket.
> por ej. feature/AP-1317-...

¿Cuál sería el valor de incluir el tipo de issue que se debe hacer?

**Comment by roanvilina** [comment]
Creo que el solo hecho de tener una plantilla para los PR nos podría ayudar bastante a explicitar más lo que se hizo y lo que se espera. Podríamos partir pronto usando la plantilla y luego de usar ver si necesita algún cambio.

**Comment by felmsena** [comment]
**Formato de título: [jira-link] Título**
En el título, lo ideal es:
-Solo dejar el ID de la tarjeta
-Completar en pocas palabras de que tratará la tarjeta (en inglés)

Hay tarjetas que toman el ID del issue y lo complementan con el título de la tarjeta, quedando tarjetas difíciles de reconocer. 

Por ejemplo **feature/AP-1209-permitir-que-la-pregunta-de-form**. 
Ahí recomiendo:
-AP-1209
-AP-1209-forms-allow-foreign-ids

Eso!

**Comment by roanvilina** [comment]
En los últimos PR he estado implementando la siguiente plantilla, podríamos cerrar en la adopción de algo como esto y dejar un tiempo para ir refinándola, hasta que enfrentemos la creación de un changelog unitario:

### Changelog / YYYY-MM-DD / [TAG-xxx]
**Scope:** las partes de la app que se anduvieron toqueteando.
#### New Features
- Aquí se detallan las funcionalidades nuevas.
#### Changes
- Aquí se detallan los cambios a funcionalidades ya existentes.
#### Bug Fixes
- Aquí se detallan arreglos a funcionalidades defectuosas.
#### Deprecations
- Aquí se detalla lo que se eliminó o se proyecta a eliminar.
#### Gems
- Aquí se detalla información sobre instalación, actualización o remoción de gemas.
#### Others
- Aquí se detallaría cualquier otra cosa.

**Comment by roanvilina** [comment]
La fecha es la fecha de la escritura del changelog, es importante para considerar que tan actual es la descripción en caso de que se hagan trabajos posteriores.

**Comment by limcross** [comment]
> En los últimos PR he estado implementando la siguiente plantilla, podríamos cerrar en la adopción de algo como esto y dejar un tiempo para ir refinándola, hasta que enfrentemos la creación de un changelog unitario:
> 
> ### Changelog / YYYY-MM-DD / [TAG-xxx]
> **Scope:** las partes de la app que se anduvieron toqueteando.
> 
> #### New Features
> * Aquí se detallan las funcionalidades nuevas.
> 
> #### Changes
> * Aquí se detallan los cambios a funcionalidades ya existentes.
> 
> #### Bug Fixes
> * Aquí se detallan arreglos a funcionalidades defectuosas.
> 
> #### Deprecations
> * Aquí se detalla lo que se eliminó o se proyecta a eliminar.
> 
> #### Gems
> * Aquí se detalla información sobre instalación, actualización o remoción de gemas.
> 
> #### Others
> * Aquí se detallaría cualquier otra cosa.

Me parece excelente, lo que sí me gustaría es incorporar un **Summary** al comienzo del changelog, que sería la forma _humano entendible_, y que podemos entregar a los clientes.

**Comment by limcross** [comment]
Ya esta disponible un template default con los elementos anteriormente mencionados 👌 

![image](https://user-images.githubusercontent.com/584752/61257829-ba7cd700-a740-11e9-94bb-80fa118768ab.png)

Me queda pendiente formalizar este issue en el gitbook.

**Comment by limcross** [comment]
ℹ️ Otro paso que debemos dar, es aplicar esta plantilla en todos nuestros repositorios.
